src/commands/registration/pasv.js

```diff
@@ -8,15 +8,18 @@ module.exports = {
       return this.reply(502);
     }
 
+    let port;
     this.connector = new PassiveConnector(this);
     return this.connector.setupServer()
     .then((server) => {
-      let address = this.server.options.pasv_url;
+      port = server.address().port;
       // Allow connecting from local
-      if (isLocalIP(this.ip)) {
-        address = this.ip;
-      }
-      const {port} = server.address();
+      if (isLocalIP(this.ip)) return this.ip;
+      let address = this.server.options.pasv_url;
```

Comment:
To avoid confusing naming shadows - I might name this something else, like `pasvAddress` or similar - that way it becomes less similar to `server.address`.

Reply:
This isn't really an issue if we structure the first promise in the chain correctly.